### PR TITLE
[stable9.1] Fix user deletion for LDAP users

### PR DESCRIPTION
**Pull request by jvillafanez**
## Description
ownCloud's user deletion fails for LDAP users who have been deleted in the LDAP server. This PR fixes that.

## Related Issue
https://github.com/owncloud/core/issues/27861

## How Has This Been Tested?
Manually tested:
* Try to delete an existing user -> failed (expected)
* Delete the user from LDAP, try to delete from ownCloud -> failed (expected, user not marked as deleteable)
* Delete the user from LDAP, check user existence (occ ldap:check-user), try to delete -> success

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

To be checked the behaviour in 10.0.x (the behaviour changed so we might need to review it), and possible backport to 9.0.x

@felixboehm @PVince81 To be evaluated if this can be in 9.1.6. Setting for 9.1.7 for now.

### CODE REVIEW

**Comment by davitol** [comment]
To test:

- [ ] Try to delete an existing user -> failed (expected)
- [ ] Delete the user from LDAP, try to delete from ownCloud -> failed (expected, user not marked as deleteable)
- [ ] Delete the user from LDAP, check user existence (occ ldap:check-user), try to delete -> success
- [ ] Having two LDAP configurations active, having a user in common,  delete the user from one of the 
LDAP, check user existence (occ ldap:check-user), try to delete

**Comment by PVince81** [comment]
@jvillafanez will this need a forward port ?

**Comment by jvillafanez** [comment]
Port to 10 isn't needed. Anyway the behaviour in 10 changed (linked issues above)

**Comment by PVince81** [comment]
@jvillafanez please rebase for CI then we merge

**Comment by jvillafanez** [comment]
rebased

File: apps/user_ldap/tests/User_LDAPTest.php [hunk 1]
```diff
@@ -290,7 +296,7 @@ public function testDeleteUserSuccess() {
 		$this->assertTrue($result);
 
 		$home = $backend->getHome('jeremy');
-		$this->assertSame($home, '/var/vhome/jdings/');
```
Comment (butonic) [inline apps/user_ldap/tests/User_LDAPTest.php]: this is problematic. AFAICT the files will no longer be deleted on disk. please double check. @davitol

**Review by butonic — COMMENTED** [review]
AFAICT the files will no longer be deleted on disk because the OfflineUser concept is now broken.

**Comment by butonic** [comment]
hm core changed in https://github.com/owncloud/core/pull/26848/files and fetches the home before deleting the user. so this might work ... I will need to set up an ldap to double check if files are really deleted.

**Review by butonic — APPROVED** [review]
nice, files are deleted as well. good work!

**Pull request merged**